src/safeposix/syscalls/net_calls.rs

```diff
@@ -159,35 +180,49 @@ impl Cage {
 
     //creates a sockhandle if none exists, otherwise this is a no-op
     pub fn force_innersocket(sockhandle: &mut SocketHandle) {
+        //If innersocket is not available, then insert a socket into innersocket
         if let None = sockhandle.innersocket {
+            //Create a new socket
```

Comment:
I can see we're creating a socket, but why?

Also, can this operation fail?   What happens if it does?